File: problems/1642-water-bottles/solution.c
int numWaterBottles(int numBottles, int numExchange) {
    int max_bottles = numBottles;
    while (numBottles / numExchange >= 1) {
        max_bottles += (numBottles / numExchange);
        numBottles = numBottles / numExchange + numBottles % numExchange;
    }
    return max_bottles;
}

File: problems/1642-water-bottles/solution.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        max_bottles = empty_bottles = numBottles
        while empty_bottles / numExchange >= 1:
            max_bottles += int(empty_bottles / numExchange)
            empty_bottles = int(empty_bottles / numExchange) + empty_bottles % numExchange
        return max_bottles

